### ISSUE TITLE
Cannot build with Snap installed cmake

### ISSUE THREAD

**Issue by hyt589** [microsoft/onnxruntime#6533]
**Describe the bug**
The build script does not work correctly when using `cmake` installed from `snap`. `os.path.realpath(shutil('cmake'))` evaluates to `/usr/bin/snap`

**Urgency**
I am trying to build onnxruntime with tensorrt provider for a company project. I can work around this issue but it involves changing the build script.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- ONNX Runtime installed from (source or binary): source
- ONNX Runtime version: v1.6.0
- Python version: 3.8.5 (miniconda)
- Visual Studio version (if applicable): none
- GCC/Compiler version (if compiling from source): 7.5.0 prebuilt
- CUDA/cuDNN version: 11.0/8.0
- GPU model and memory: RTX6000-4Q, 4GiB

Note: this is a virtual machine.

**To Reproduce**
- Describe steps/code to reproduce the behavior.
  1. make a new virtual machine or uninstall `cmake` from system
  2. install `cmake` with `snap` and add it to `PATH`
  3. run the script with the following parameters:
```sh
./build.sh --config Release --parallel --cuda_home /usr/local/cuda --cudnn_home /usr/local/cuda --use_tensorrt --tensorrt_home /home/hyt/resources/TensorRT-7.2.2.3 --cmake_path /snap/bin/cmake
```
  4. it complains that `snap` followed by the cmake directory is not a valid command:
```
error: unknown command "/home/hyt/workspace/onnxruntime/cmake", see 'snap help'.
Traceback (most recent call last):
  File "/home/hyt/workspace/onnxruntime/tools/ci_build/build.py", line 2088, in <module>
    sys.exit(main())
  File "/home/hyt/workspace/onnxruntime/tools/ci_build/build.py", line 2008, in main
    generate_build_tree(
  File "/home/hyt/workspace/onnxruntime/tools/ci_build/build.py", line 1006, in generate_build_tree
    run_subprocess(
  File "/home/hyt/workspace/onnxruntime/tools/ci_build/build.py", line 523, in run_subprocess
    return run(*args, cwd=cwd, capture_stdout=capture_stdout, shell=shell, env=my_env)
  File "/home/hyt/workspace/onnxruntime/tools/python/util/run.py", line 39, in run
    completed_process = subprocess.run(
  File "/home/hyt/miniconda3/lib/python3.8/subprocess.py", line 512, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/bin/snap', '/home/hyt/workspace/onnxruntime/cmake', '-Donnxruntime_RUN_ONNX_TESTS=OFF', '-Donnxruntime_BUILD_WINML_TESTS=ON', '-Donnxruntime_GENERATE_TEST_REPORTS=ON', '-Donnxruntime_DEV_MODE=ON', '-DPYTHON_EXECUTABLE=/home/hyt/miniconda3/bin/python3', '-Donnxruntime_USE_CUDA=ON', '-Donnxruntime_CUDNN_HOME=/usr/local/cuda', '-Donnxruntime_USE_FEATURIZERS=OFF', '-Donnxruntime_CUDA_HOME=/usr/local/cuda', '-Donnxruntime_USE_JEMALLOC=OFF', '-Donnxruntime_USE_MIMALLOC_STL_ALLOCATOR=OFF', '-Donnxruntime_USE_MIMALLOC_ARENA_ALLOCATOR=OFF', '-Donnxruntime_ENABLE_PYTHON=OFF', '-Donnxruntime_BUILD_CSHARP=OFF', '-Donnxruntime_BUILD_JAVA=OFF', '-Donnxruntime_BUILD_NODEJS=OFF', '-Donnxruntime_BUILD_SHARED_LIB=OFF', '-Donnxruntime_USE_EIGEN_FOR_BLAS=ON', '-Donnxruntime_USE_OPENBLAS=OFF', '-Donnxruntime_USE_DNNL=OFF', '-Donnxruntime_DNNL_GPU_RUNTIME=', '-Donnxruntime_DNNL_OPENCL_ROOT=', '-Donnxruntime_USE_MKLML=OFF', '-Donnxruntime_USE_NNAPI_BUILTIN=OFF', '-Donnxruntime_USE_RKNPU=OFF', '-Donnxruntime_USE_OPENMP=OFF', '-Donnxruntime_USE_TVM=OFF', '-Donnxruntime_USE_LLVM=OFF', '-Donnxruntime_ENABLE_MICROSOFT_INTERNAL=OFF', '-Donnxruntime_USE_VITISAI=OFF', '-Donnxruntime_USE_NUPHAR=OFF', '-Donnxruntime_USE_TENSORRT=ON', '-Donnxruntime_TENSORRT_HOME=/home/hyt/resources/TensorRT-7.2.2.3', '-Donnxruntime_USE_MIGRAPHX=OFF', '-Donnxruntime_MIGRAPHX_HOME=', '-Donnxruntime_CROSS_COMPILING=OFF', '-Donnxruntime_DISABLE_CONTRIB_OPS=OFF', '-Donnxruntime_DISABLE_ML_OPS=OFF', '-Donnxruntime_DISABLE_RTTI=OFF', '-Donnxruntime_DISABLE_EXCEPTIONS=OFF', '-Donnxruntime_DISABLE_ORT_FORMAT_LOAD=OFF', '-Donnxruntime_MINIMAL_BUILD=OFF', '-Donnxruntime_EXTENDED_MINIMAL_BUILD=OFF', '-Donnxruntime_REDUCED_OPS_BUILD=OFF', '-Donnxruntime_MSVC_STATIC_RUNTIME=OFF', '-Donnxruntime_ENABLE_LANGUAGE_INTEROP_OPS=OFF', '-Donnxruntime_USE_DML=OFF', '-Donnxruntime_USE_WINML=OFF', '-Donnxruntime_USE_TELEMETRY=OFF', '-Donnxruntime_ENABLE_LTO=OFF', '-Donnxruntime_USE_ACL=OFF', '-Donnxruntime_USE_ACL_1902=OFF', '-Donnxruntime_USE_ACL_1905=OFF', '-Donnxruntime_USE_ACL_1908=OFF', '-Donnxruntime_USE_ACL_2002=OFF', '-Donnxruntime_USE_ARMNN=OFF', '-Donnxruntime_ARMNN_RELU_USE_CPU=ON', '-Donnxruntime_ARMNN_BN_USE_CPU=ON', '-Donnxruntime_ENABLE_NVTX_PROFILE=OFF', '-Donnxruntime_ENABLE_TRAINING=OFF', '-Donnxruntime_USE_HOROVOD=OFF', '-Donnxruntime_USE_NCCL=ON', '-Donnxruntime_BUILD_BENCHMARKS=OFF', '-Donnxruntime_USE_ROCM=OFF', '-Donnxruntime_ROCM_HOME=', '-Donnxruntime_USE_FULL_PROTOBUF=ON', '-Donnxruntime_USE_FULL_PROTOBUF=ON', '-DProtobuf_USE_STATIC_LIBS=ON', '-DCUDA_CUDA_LIBRARY=/usr/local/cuda/lib64/stubs', '-Donnxruntime_PYBIND_EXPORT_OPSCHEMA=OFF', '-Donnxruntime_ENABLE_MEMLEAK_CHECKER=OFF', '-DCMAKE_BUILD_TYPE=Release']' returned non-zero exit status 64.
```
- Attach the ONNX model to the issue (where applicable) to expedite investigation.
  - There is no model involved in this issue

**Expected behavior**
The build script should be able to build the project with a snap installed `cmake`.

**Comment by snnn** [comment]
I think it is due to #205 

@edgchen1  Could you please take a look?

**Comment by edgchen1** [comment]
Should be fixed by #6540. @hyt589 please re-open if it's still not working.
